project/MiMa.scala

```diff
@@ -0,0 +1,81 @@
+// It would be nice to use sbt-mima-plugin here, but the plugin is missing
+// at least two features we need:
+// * ability to run MiMa twice, swapping `curr` and `prev`, to detect
+//   both forwards and backwards incompatibilities
+// * ability to pass a filter file
+// So we invoke the MiMa CLI directly; it's also what the Ant build did.
+
+import sbt._
+import sbt.Keys._
+
+object MiMa {
+
+  lazy val mimaOldVersion =
+    settingKey[String]("Scala version number to run MiMa against")
+
+  lazy val mima =
+    taskKey[Unit]("run Migration Manager to detect binary incompatibilities")
+
+  lazy val settings =
+    Seq(
+      mimaOldVersion := s"${scalaBinaryVersion.value}.0",
+      mima := {
+        val log = streams.value.log
+        def runOnce(prev: java.io.File, curr: java.io.File, isForward: Boolean): Unit = {
+          val direction = if (isForward) "forward" else "backward"
+          log.info(s"Checking $direction binary compatibility")
+          runMima(
+            prev = if (isForward) curr else prev,
+            curr = if (isForward) prev else curr,
+            // TODO: it would be nicer if each subproject had its own whitelist, but for now
+            // for compatibility with how Ant did things, there's just one at the root.
+            // once Ant is gone we'd be free to split it up.
```

Comment:
Once ANT is gone, I'd just put it into build.sbt and get rid of the extra files.